pkg/prometheus/promcfg_test.go

```diff
@@ -12051,3 +12061,89 @@ func TestAlertmanagerTLSConfig(t *testing.T) {
 
 	}
 }
+
+func TestServiceMonitorSelectors(t *testing.T) {
```

Comment:
can we add a test with a Prometheus resource which has `.spec.serviceDiscoveryRole: EndpointSlice`? It should generate `role: endpointslice` rather than endpoints.

Reply:
I'll move away from endpoints and endpoints slices and only use service role, since we're already filtering by service labels, this will simplify tests and ensure we're operate over the already used labels

Reply:
I think that it makes sense to filter both services and endpoints(lice). Otherwise we're not meeting the user expectations (reducing the number of resources being watched).